### PR TITLE
Fix Bugs, Update JavaDocs and Include PERT Chart

### LINKED ISSUE #115: Bug when list-n /tag /sort
NullPointerException thrown in code. Refer to snippets below

image

Cause: Boolean isPinned is null

image

### PR DESCRIPTION
**Pull request by R-Ramana**
Closes #115 

1. Updated JavaDocs for some methods for the following Classes:

- ArchiveNoteCommand
- ListNoteCommand
- Note
- Notebook
- Parser
- UnarchiveNoteCommand

2. Fixed bug for list-n /tag /sort 
3. Helped update the getPinnedString() in the PinCommand and StorageManager classes.

### CODE REVIEW

File: docs/diagrams/PERT_Chart.puml [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        2 MD
+    ]
+    Rectangle Brandon_1 [
+        Brandon
+        ....
+        Event Function
+        ====
+        Add Daily/Weekly Events
+        ---
+        2 MD
```
Comment (Chongjx) [inline docs/diagrams/PERT_Chart.puml:34]:
```suggestion
        3 MD
```

File: docs/diagrams/PERT_Chart.puml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        ----
+        2 MD
+    ]
+    Rectangle JunXiang_2 [
+        Jun Xiang
+        ....
+        Tag Function
+        ====
+        Tag Events
+        Alert on Clashing Events
```
Comment (Chongjx) [inline docs/diagrams/PERT_Chart.puml:71]:
```suggestion
```

File: docs/diagrams/PERT_Chart.puml [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        ----
+        1 MD
```
Comment (Chongjx) [inline docs/diagrams/PERT_Chart.puml:73]:
```suggestion
        2 MD
```

File: docs/diagrams/PERT_Chart.puml [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        1.5 MD
+    ]
+    Rectangle JunXiang_3 [
+        Jun Xiang
+        ....
+        Tag Function
+        ====
+        Add Color
+        Tag Notes
+    ]
```
Comment (Chongjx) [inline docs/diagrams/PERT_Chart.puml:93]:
```suggestion
        Tag Notes
        ----
        5 MD
    ]
```

File: docs/diagrams/PERT_Chart.puml [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    Rectangle Nazryl_1 [
+        Nazryl
+        ....
+        Note Function
+        ====
+        Add Note
+        Delete Note
+        ----
+        1 MD
```
Comment (Chongjx) [inline docs/diagrams/PERT_Chart.puml:53]:
```suggestion
        5 MD
```

File: docs/diagrams/PERT_Chart.puml [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+        2 MD
+    ]
+    Rectangle Brandon_3 [
+        Brandon
+        ....
+        Event Function
+        ====
+        Edit Events
+        ----
+        1 MD
```
Comment (Chongjx) [inline docs/diagrams/PERT_Chart.puml:114]:
```suggestion
        2 MD
```

File: docs/diagrams/PERT_Chart.puml [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+    ]
+    Rectangle Prachi_2 [
+        Prachi
+        ....
+        Storage Function
+        ====
+        Import Events
+        Export Events
+        ----
+        1 MD
```
Comment (Chongjx) [inline docs/diagrams/PERT_Chart.puml:124]:
```suggestion
        2 MD
```

File: docs/diagrams/PERT_Chart.puml [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+    ]
+    Rectangle Prachi_3 [
+        Prachi
+        ....
+        Storage Function
+        ====
+        Import Notes
+        Export Notes
+        ----
+        1 MD
```
Comment (Chongjx) [inline docs/diagrams/PERT_Chart.puml:134]:
```suggestion
        2 MD
```

File: docs/diagrams/PERT_Chart.puml [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+        1.5 MD
+    ]
+    Rectangle Nazryl_2 [
+        Nazryl
+        ....
+        Note Function
+        ====
+        Edit Notes
+        ----
+        1 MD
```
Comment (Chongjx) [inline docs/diagrams/PERT_Chart.puml:152]:
```suggestion
        2 MD
```

**Review by Chongjx — APPROVED** [review]
I mainly only looked at the Pert Chart. The rest of the stuff looks good to me.

File: docs/diagrams/PERT_Chart.puml [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+    ]
+    Rectangle Ramana_4 [
+        Ramana
+        ....
+        Archive Function
+        ====
+        Archive Notes
+        ----
+        1.5 MD
```
Comment (R-Ramana) [inline docs/diagrams/PERT_Chart.puml:144]:
```suggestion
        3 MD
```

File: docs/diagrams/PERT_Chart.puml [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+    Rectangle Ramana_3 [
+        Ramana
+        ....
+        List Function
+        Find Function
+        ====
+        List Notes
+        Find Notes
+        ----
+        1.5 MD
```
Comment (R-Ramana) [inline docs/diagrams/PERT_Chart.puml:83]:
```suggestion
        3 MD
```

**Pull request merged**